packages/jest-core/src/runJest.ts

```diff
@@ -192,6 +192,9 @@ export default async function runJest({
     }),
   );
 
+  allTests = globalConfig.shard
+    ? sequencer.shard(allTests, globalConfig.shard)
```

Comment:
I think this will just throw if not defined? We should make this explicit.

```ts
if (globalConfig.shard) {
  if (typeof sequencer.shard !== 'function') {
    throw new Error(
      `"shard" defined on options, but the test scheduler at ${pathToScheduler} does not support it`,
    );
  }

  allTests = sequencer.shard(allTests, globalConfig.shard);
}
```

or something like that 🙂